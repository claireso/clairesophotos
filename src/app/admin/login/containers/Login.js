import { connect } from 'react-redux'
import { navigate } from '@reach/router'

import Login from '../views/Login'

import {
  login,
  LOGIN_SUCCESS,
  loadUser,
  LOAD_USER_SUCCESS
} from '../../../../common/actions/user'

import { UNAUTHORIZED_ERROR } from '../../../../common/actions/api'

const mapDispatchToProps = dispatch => ({
  login(data) {
    dispatch(login(data)).then(action => {
      if (action.type === LOGIN_SUCCESS) {
        dispatch(loadUser()).then(action => {
          if (action.type === LOAD_USER_SUCCESS) {
            //@TODO: next parameter in url
            navigate('/admin/photos')
          }
        })
      }

      if (action.type === UNAUTHORIZED_ERROR) {
        alert('Bad username/password. Please retry')
      }
    })
  }
})

export default connect(
  null,
  mapDispatchToProps
)(Login)
