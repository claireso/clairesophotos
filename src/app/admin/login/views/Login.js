import React from 'react'
import PropTypes from 'prop-types'

import { Heading1 } from '@admin/components/Headings'
import Box from '@admin/components/Box'

import Form from './form/Form'

const Login = props => {
  return (
    <main>
      <Box>
        <Heading1>Login</Heading1>
        <Form onSubmit={props.login} isProcessing={props.isLogin} />
      </Box>
    </main>
  )
}

Login.propTypes = {
  login: PropTypes.func.isRequired,
  isLogin: PropTypes.bool.isRequired
}

export default Login
