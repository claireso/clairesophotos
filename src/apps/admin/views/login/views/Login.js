import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { Heading1 } from '@admin/components/Headings'
import Box from '@admin/components/Box'

import Form from './form/Form'

const Wrapper = styled.div`
  align-items: center;
  display: flex;
  height: 100vh;

  > main {
    width: 100%;
    padding-top: 0;
    padding-bottom: 0;
  }
`

const Login = props => {
  return (
    <Wrapper>
      <main>
        <Box>
          <Heading1>Login</Heading1>
          <Form onSubmit={props.login} isProcessing={props.isLogin} />
        </Box>
      </main>
    </Wrapper>
  )
}

Login.defaultProps = {
  isLogin: false
}

Login.propTypes = {
  login: PropTypes.func.isRequired,
  isLogin: PropTypes.bool.isRequired
}

export default Login
