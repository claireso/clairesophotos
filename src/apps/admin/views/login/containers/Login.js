import { connect } from 'react-redux'
import { navigate } from '@reach/router'

import Login from '../views/Login'

import {
  login,
  LOGIN_SUCCESS,
  loadUser,
  LOAD_USER_SUCCESS
} from '@admin/actions/user'

import { UNAUTHORIZED_ERROR } from '@admin/actions/user'

const mapStateToProps = state => ({
  isLogin: state.user.isLogin
})

const mapDispatchToProps = dispatch => ({
  login(data) {
    dispatch(login(data)).then(action => {
      if (action.type === LOGIN_SUCCESS) {
        dispatch(loadUser()).then(action => {
          if (action.type === LOAD_USER_SUCCESS) {
            const parsedUrl = new URL(window.location.href)
            const next = parsedUrl.searchParams.get('next')

            navigate(next || '/admin/photos')
          }
        })
      }

      if (action.type === UNAUTHORIZED_ERROR) {
        alert('Bad username/password. Please retry')
      }
    })
  }
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Login)
