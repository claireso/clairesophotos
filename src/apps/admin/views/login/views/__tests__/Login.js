import React from 'react'
import { render, fireEvent } from 'react-testing-library'

import Login from '../Login'

describe('<Login />', () => {
  test('should render component', () => {
    const { container } = render(<Login login={() => {}} />)

    expect(container).toMatchSnapshot()
  })

  test('should render component in processing status', () => {
    const { container } = render(<Login login={() => {}} isLogin />)

    expect(container).toMatchSnapshot()
  })

  test('should call login function', () => {
    const spyOnSubmit = jest.fn()

    const { getByLabelText, getByDisplayValue } = render(
      <Login login={spyOnSubmit} />
    )

    fireEvent.change(getByLabelText(/username/i), {
      target: { value: 'admin' }
    })
    fireEvent.change(getByLabelText(/password/i), {
      target: { value: 'password' }
    })

    fireEvent.click(getByDisplayValue(/log in/i))

    expect(spyOnSubmit).toHaveBeenCalledWith({
      username: 'admin',
      password: 'password'
    })
  })
})
