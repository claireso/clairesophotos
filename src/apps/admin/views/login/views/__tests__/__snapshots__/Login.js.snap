// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render component 1`] = `
.c2 {
  font-size: 2.4rem;
  margin: 0 0 2.4rem;
}

.c1 {
  background: var(--white);
  border-radius: 0.4rem;
  border: 1px solid var(--gray-1);
  box-shadow: 0px 1px 5px var(--box-shadow);
  margin: 0 auto;
  max-width: 40rem;
  padding: calc(var(--gutter) * 2);
}

.c3 {
  margin: 0 0 var(--gutter);
}

.c4 {
  display: block;
  font-size: 1.6rem;
  line-height: 1.5;
  margin: 0 0 1.2rem;
}

.c5 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid var(--gray-1);
  border-radius: 0.4rem;
  box-shadow: 1px 1px 3px var(--box-shadow);
  display: block;
  outline: none;
  padding: 0.8rem 1.2rem;
  width: 100%;
  font-size: 1.4rem;
  line-height: 1.71;
  -webkit-transition: border-color 250ms ease-out;
  transition: border-color 250ms ease-out;
}

.c5:focus {
  border-color: var(--gray-2);
}

.c6 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: var(--primary);
  border: none;
  border-radius: 3.6rem;
  color: white;
  cursor: pointer;
  display: block;
  font-size: 1.4rem;
  line-height: 1;
  margin: 2.8rem auto 0;
  outline: none;
  padding: 1.6rem 5.2rem;
  -webkit-transition: background 150ms ease-out;
  transition: background 150ms ease-out;
}

.c6:hover {
  background: var(--primary-lighter);
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100vh;
}

.c0 > main {
  width: 100%;
  padding-top: 0;
  padding-bottom: 0;
}

<div>
  <div
    class="c0"
  >
    <main>
      <div
        class="c1"
      >
        <h1
          class="c2"
        >
          Login
        </h1>
        <form
          method="POST"
        >
          <div
            class="c3"
          >
            <label
              class="c4"
              for="username"
            >
              Username
            </label>
            <input
              class="c5"
              id="username"
              name="username"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="c3"
          >
            <label
              class="c4"
              for="password"
            >
              Password
            </label>
            <input
              class="c5"
              id="password"
              name="password"
              required=""
              type="password"
              value=""
            />
          </div>
          <input
            class="c6"
            type="submit"
            value="Log in"
          />
        </form>
      </div>
    </main>
  </div>
</div>
`;

exports[`<Login /> should render component in processing status 1`] = `
.c2 {
  font-size: 2.4rem;
  margin: 0 0 2.4rem;
}

.c1 {
  background: var(--white);
  border-radius: 0.4rem;
  border: 1px solid var(--gray-1);
  box-shadow: 0px 1px 5px var(--box-shadow);
  margin: 0 auto;
  max-width: 40rem;
  padding: calc(var(--gutter) * 2);
}

.c3 {
  margin: 0 0 var(--gutter);
}

.c4 {
  display: block;
  font-size: 1.6rem;
  line-height: 1.5;
  margin: 0 0 1.2rem;
}

.c5 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid var(--gray-1);
  border-radius: 0.4rem;
  box-shadow: 1px 1px 3px var(--box-shadow);
  display: block;
  outline: none;
  padding: 0.8rem 1.2rem;
  width: 100%;
  font-size: 1.4rem;
  line-height: 1.71;
  -webkit-transition: border-color 250ms ease-out;
  transition: border-color 250ms ease-out;
}

.c5:focus {
  border-color: var(--gray-2);
}

.c8 {
  margin: 2rem 0;
  -webkit-animation: jitEcr 800ms ease-in-out infinite alternate;
  animation: jitEcr 800ms ease-in-out infinite alternate;
}

.c8:after {
  background: var(--primary);
  border-radius: 50%;
  content: '';
  display: block;
  height: 1rem;
  width: 1rem;
  margin: 0 auto;
}

.c6 {
  margin: 2.8rem 0 0;
  padding: 1.8rem 0 1.8rem;
}

.c6 .c7 {
  margin: 0 auto;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100vh;
}

.c0 > main {
  width: 100%;
  padding-top: 0;
  padding-bottom: 0;
}

<div>
  <div
    class="c0"
  >
    <main>
      <div
        class="c1"
      >
        <h1
          class="c2"
        >
          Login
        </h1>
        <form
          method="POST"
        >
          <div
            class="c3"
          >
            <label
              class="c4"
              for="username"
            >
              Username
            </label>
            <input
              class="c5"
              id="username"
              name="username"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="c3"
          >
            <label
              class="c4"
              for="password"
            >
              Password
            </label>
            <input
              class="c5"
              id="password"
              name="password"
              required=""
              type="password"
              value=""
            />
          </div>
          <div
            class="c6"
          >
            <div
              class="c7 c8"
            />
          </div>
        </form>
      </div>
    </main>
  </div>
</div>
`;
